### ISSUE TITLE
[Feature]: Allow creating mock API endpoints

### ISSUE THREAD

**Issue by sumitsum** [appsmithorg/appsmith#10609]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

Provide a feature that allows creating mock API endpoins. 

### Why should this be worked on?

API is one of the most used datasources in Appsmith. Users typically want to bind their data to widgets or write JS code around it. In case the actual API endpoint is not available or unusable or under development then users can create a mock API endpoint to return a sample data and build or test their applications around it.

**Comment by Rishabhkaul** [comment]
Is this also a requirement for templates to work well @GreenFlux ?

**Comment by GreenFlux** [comment]
@Rishabhkaul yes this would allow us to produce a wider range of templates, and enable us to mock popular SaaS APIs instead of having to maintain an account and API keys for each SaaS. 

I brought up using Mockoon before but @sharat87 looked into it and said there are some challenges with hosting it and exposing the API publicly.
